addon/utils/query-params.ts

```diff
@@ -113,8 +112,8 @@ export function getParamsObject({
             }
 
             let value = get(context, valueKey);
-            if (moment.isMoment(value) || moment.isDate(value)) {
-                value = moment(value).format(serverDateFormat);
+            if (value instanceof Date) {
+                value = serializeDate(value);
             }
```

Comment:
So this isn't gonna work when the app overrides the serialize function to use luxon or something, because they will have Luxon date object's not `Date`'s.

What we probably need to do is just generalize this serialize function, (and let it accept an arg of `any` type) and call it on every value , and then you move this type checking logic into the serialize function itself. e.g.

```ts
const value = serializeFilterValue(key, get(context, valueKey));
```
and the new default implementation of the serialize function would look like:
```ts
serializeFilterValue(key: string, value: any) => {
    if (value typeof Date) {
        return value.toISOString();
    }
    
    return value;
}
```
(passing in `key` ^ in there would also allow apps to do serialization logic based on the param name too if they needed, instead of only based on the value type)